### PR TITLE
Improve management of CAPI images

### PR DESCRIPTION
**Pull request by martinmo**
Hi, as part of https://github.com/SovereignCloudStack/standards/pull/643, I want to make it as easy as possible to get the latest CAPI images in an SCS compliant way and found `osism manage image clusterapi` to be the best way to do it, with the following adjustments.

(Potentially) breaking changes:

* Changed the absolute invocation of `openstack-image-manager`, to make it work if installed in other places. So now it depends on `PATH`. I *think* this shouldn't hurt. (Actually, it will even improve things in case someone installs stuff into a virtualenv, for example.)
* Changed image name `Kubernetes CAPI` to `ubuntu-capi-image` as recommended by SCS.

Otherwise it just fixes bugs (hard-coded `--cloud`) and adds new features (`--dry-run`, `--tag`) in a backwards compatible way.

Resulting help output:

```
$ osism manage image clusterapi --help                                      
[... some unrelated TripleDES deprecation warnings ...]
usage: osism manage image clusterapi [-h] [--base-url BASE_URL] [--cloud CLOUD] [--dry-run] [--tag TAG] [--filter FILTER]

options:
  -h, --help            show this help message and exit
  --base-url BASE_URL
                        Base URL
  --cloud CLOUD
                        Cloud name in clouds.yaml (will be overruled by OS_AUTH_URL envvar)
  --dry-run             Do not perform any changes (--dry-run passed to openstack-image-manager)
  --tag TAG     Name of the tag used to identify managed images (use openstack-image-manager's default if unset)
  --filter FILTER
                        Filter the version to be managed (e.g. 1.28)
```

Example invocation:

```
$ osism manage image clusterapi --dry-run --cloud l1a --tag 'managed_by_C&H'
[... some unrelated TripleDES deprecation warnings ...]
2024-08-07 12:34:06 | INFO     | date: 2024-07-18
2024-08-07 12:34:06 | INFO     | image: ubuntu-2204-kube-v1.28/ubuntu-2204-kube-v1.28.12.qcow2
2024-08-07 12:34:06 | INFO     | version: 1.28.12
2024-08-07 12:34:06 | INFO     | url: https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube-v1.28/ubuntu-2204-kube-v1.28.12.qcow2
2024-08-07 12:34:06 | INFO     | checksum_url: https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube-v1.28/ubuntu-2204-kube-v1.28.12.qcow2.CHECKSUM
2024-08-07 12:34:06 | INFO     | checksum: 45c75763bdb53c17d3482abd6d0a55b2429efd7df7927fd2a1d68c4cf859b146
2024-08-07 12:34:06 | INFO     | date: 2024-07-18
2024-08-07 12:34:06 | INFO     | image: ubuntu-2204-kube-v1.29/ubuntu-2204-kube-v1.29.7.qcow2
2024-08-07 12:34:06 | INFO     | version: 1.29.7
2024-08-07 12:34:06 | INFO     | url: https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube-v1.29/ubuntu-2204-kube-v1.29.7.qcow2
2024-08-07 12:34:06 | INFO     | checksum_url: https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube-v1.29/ubuntu-2204-kube-v1.29.7.qcow2.CHECKSUM
2024-08-07 12:34:07 | INFO     | checksum: 35d3f1e124c0a3f428487d2266923bbed066f87da78ad45245075cb00cfbd33f
2024-08-07 12:34:07 | INFO     | date: 2024-07-18
2024-08-07 12:34:07 | INFO     | image: ubuntu-2204-kube-v1.30/ubuntu-2204-kube-v1.30.3.qcow2
2024-08-07 12:34:07 | INFO     | version: 1.30.3
2024-08-07 12:34:07 | INFO     | url: https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube-v1.30/ubuntu-2204-kube-v1.30.3.qcow2
2024-08-07 12:34:07 | INFO     | checksum_url: https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube-v1.30/ubuntu-2204-kube-v1.30.3.qcow2.CHECKSUM
2024-08-07 12:34:07 | INFO     | checksum: c248a2def6110e65dd93c7fb8fd3d9d51904efce7aa54f0581200339ef849386
2024-08-07 12:34:09 | INFO     | Processing image 'ubuntu-capi-image 1.30.3'
2024-08-07 12:34:09 | INFO     | Tested URL https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube-v1.30/ubuntu-2204-kube-v1.30.3.qcow2: 200
2024-08-07 12:34:09 | INFO     | Skipping required import of image 'ubuntu-capi-image 1.30.3', running in dry-run mode
2024-08-07 12:34:09 | INFO     | Processing image 'ubuntu-capi-image 1.29.7'
2024-08-07 12:34:09 | INFO     | Tested URL https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube-v1.29/ubuntu-2204-kube-v1.29.7.qcow2: 200
2024-08-07 12:34:09 | INFO     | Skipping required import of image 'ubuntu-capi-image 1.29.7', running in dry-run mode
2024-08-07 12:34:09 | INFO     | Processing image 'ubuntu-capi-image 1.28.12'
2024-08-07 12:34:10 | INFO     | Tested URL https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube-v1.28/ubuntu-2204-kube-v1.28.12.qcow2: 200
2024-08-07 12:34:10 | INFO     | Skipping required import of image 'ubuntu-capi-image 1.28.12', running in dry-run mode
```

### CODE REVIEW

File: osism/commands/manage.py [hunk 1]
```diff
@@ -41,6 +52,7 @@ def take_action(self, parsed_args):
         base_url = parsed_args.base_url
         cloud = parsed_args.cloud
         filter = parsed_args.filter
```
[the remark this answers is not in the document]
Comment (berendt) [reply osism/commands/manage.py:57]: This probably makes sense and can be tackled In a separate PR.

**Comment by berendt** [comment]
@martinmo The change of the image name is required because of the SCS standard?

**Comment by martinmo** [comment]
@berendt Yes, in https://github.com/SovereignCloudStack/standards/blob/2d7f7d31ad5612c9db0699a626e2eace4e41c383/Tests/iaas/scs-0104-v1-images.yaml#L7-L10 we use this name scheme:

```yaml
- name: "ubuntu-capi-image"
  name_scheme: "ubuntu-capi-image v[0-9]\\.[0-9]+(\\.[0-9]+)?"
  source: https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube
  status: recommended
```

Of course, it says recommended and tests won't fail if there are only `Kubernetes CAPI` images, but I don't see why someone would not use the recommended name. Also, the KaaS reference implementation is supposed to use this name, albeit there is a small inconcistency at the moment which prevents that (https://github.com/SovereignCloudStack/cluster-stacks/issues/156).

If you disagree, I could also add another flag `--custom-name` to set a custom name and let it default to the existing value `Kubernetes CAPI`. And in the implementation notes for the standard, I'll refer users to use `--custom-name ubuntu-capi-image`.

**Comment by martinmo** [comment]
Just noticed that my changes to make the image names were not complete: versions need to be prefixed with `v` as in `v1.29.3`. I adjusted the template accordingly.

Example dry run:

```
$ rm -r /tmp/clusterapi 
$ osism manage image clusterapi --dry-run --cloud l1a --filter 1.30
[... some unrelated TripleDES deprecation warnings ...]
2024-08-07 19:39:01 | INFO     | date: 2024-07-18
2024-08-07 19:39:01 | INFO     | image: ubuntu-2204-kube-v1.30/ubuntu-2204-kube-v1.30.3.qcow2
2024-08-07 19:39:01 | INFO     | version: 1.30.3
2024-08-07 19:39:01 | INFO     | url: https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube-v1.30/ubuntu-2204-kube-v1.30.3.qcow2
2024-08-07 19:39:01 | INFO     | checksum_url: https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube-v1.30/ubuntu-2204-kube-v1.30.3.qcow2.CHECKSUM
2024-08-07 19:39:01 | INFO     | checksum: c248a2def6110e65dd93c7fb8fd3d9d51904efce7aa54f0581200339ef849386
2024-08-07 19:39:03 | INFO     | Processing image 'ubuntu-capi-image v1.30.3'
2024-08-07 19:39:03 | INFO     | Tested URL https://swift.services.a.regiocloud.tech/swift/v1/AUTH_b182637428444b9aa302bb8d5a5a418c/openstack-k8s-capi-images/ubuntu-2204-kube-v1.30/ubuntu-2204-kube-v1.30.3.qcow2: 200
2024-08-07 19:39:03 | INFO     | Skipping required import of image 'ubuntu-capi-image v1.30.3', running in dry-run mode
```

**Pull request merged**